README.md

```diff
@@ -263,15 +289,46 @@ is required over query string. It may need another queryFn if the pagination tok
 }
 ```
 
-An alternative to using the `getNextPageParam` or `getPreviousPageParam` callback options, the query methods also accept a `pageParam` input.
+### `useInfiniteAPIQuery`
 
-```typescript
-const lastPage = query?.data?.pages[query.data.pages.length - 1];
-query.fetchNextPage({
-  pageParam: {
-    after: lastPage?.next,
+Type-safe wrapper around [`useInfiniteQuery`](https://tanstack.com/query/latest/docs/react/reference/useInfiniteQuery) from `react-query` which has a similar api as `useQuery` with a few key differences.
+
+```tsx
+const query = useInfiniteAPIQuery(
+  'GET /list',
+  {
+    // after is the token name in query string for the next page to return.
+    after: undefined,
```

Comment:
Is this optional when there isn't one available?

Reply:
its literally just the typed input for the list endpoint - in this example we use `after` but it could be something like `pageSize: number, page: number` or whatever

Reply:
I think I just thought it was weird that we'd pass `after: undefined` instead of just leaving it out. If the API could support just leaving it out it'd be much nicer.